### PR TITLE
[FIXED] JetStream: stream blocked recovering snapshot

### PR DESCRIPTION
**Pull request by kozlovic**
If a node fell behind, when catching up with the rest of the
cluster, it is possible that a lot of append entries accumulate
and the server would print warnings such as:
```
[WRN] RAFT [jZ6RvVRH - S-R3F-CQw2ImK6] <some number> append entries pending
```
It would then continously print the following warning:
```
AppendEntry failed to be placed on internal channel
```
When that happens, this node would always be shown with be running the
same number of operations behind (using `nats s info`) if there are
no new messages added to the stream, or an increasing number of
operations if there is still activity.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/raft.go [hunk 1]
```diff
@@ -263,6 +263,7 @@ var (
 	errLeaderLen         = fmt.Errorf("raft: leader should be exactly %d bytes", idLen)
 	errTooManyEntries    = errors.New("raft: append entry can contain a max of 64k entries")
 	errBadAppendEntry    = errors.New("raft: append entry corrupt")
+	errChannelFull       = errors.New("raft: entry failed to be placed on internal channel")
```
Comment (derekcollison) [inline server/raft.go:266]: Are there other places we could use this? I think this code has a few select blocks on channels IIRC.

**Review by derekcollison — APPROVED** [review]
LGTM - One comment on other places we might be able to use this.

File: server/raft.go (hunk 1 above)
Comment (kozlovic) [reply server/raft.go:266]: In this file, we either use a different error, such as `errProposalFailed`, or simply n.error()/n.warn()/n.debug() in the "default:" case. Following our discussion about possibly removing use of channels, I think this is a stop gap and this error will disappear in the future.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**